plugin/evm/vm.go

```diff
@@ -369,6 +370,11 @@ func (vm *VM) Initialize(
 	vm.chainConfig = g.Config
 	vm.networkID = vm.ethConfig.NetworkId
 
+	// check that subnetEVM upgrade is enabled from genesis before upgradeBytes
+	if !vm.chainConfig.IsSubnetEVM(common.Big0) {
+		return errSubnetEVMUpgradeNotEnabled
+	}
+
```

Comment:
Shall we move this below to` upgradeBytes` parsing, in case this was activated by a upgrade at genesis? Could this be a possibility? @aaronbuchwald @darioush

Reply:
They could include it in the upgrade bytes to activate at timestamp 0, but it SHOULD always be included in the genesis, so I think it's preferable to error here rather than after applying the `upgradeBytes`. I actually asked @anusha-ctrl to make this change to put it here.

Reply:
just thinking someone might want to fix their genesis via this method (or by removing the checks via config flag).